module/signature/random_beacon_signer_store.go

```diff
@@ -0,0 +1,109 @@
+package signature
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/storage"
+)
+
+type EpochAwareRandomBeaconSignerStore struct {
+	epochLookup module.EpochLookup          // used to fetch epoch counter by view
+	keys        storage.DKGKeys             // used to fetch DKG private key by epoch
+	signers     map[uint64]module.MsgSigner // cache of signers by epoch
+}
+
+func NewEpochAwareRandomBeaconSignerStore(epochLookup module.EpochLookup, keys storage.DKGKeys) *EpochAwareRandomBeaconSignerStore {
+	return &EpochAwareRandomBeaconSignerStore{
+		epochLookup: epochLookup,
+		keys:        keys,
+		signers:     make(map[uint64]module.MsgSigner),
+	}
+}
+
+// GetSigner returns the threshold-signer for signing objects at a
```

Comment:
Since we've decided to go for a stateless signature, it doesn't make sense to speak about threshold signing here. From the perspective of a non-leader node, it is just generating a simple signature using a private key.

```suggestion
// GetSigner returns the random beacon signer for signing objects at a
```